packages/block-library/src/image/edit.native.js

```diff
@@ -338,6 +336,49 @@ export class ImageEdit extends React.Component {
 			: width;
 	}
 
+	getLinkSettings() {
+		const { isLinkSheetVisible } = this.state;
+		const {
+			attributes: { href: url, ...unMappedAttributes },
+			setAttributes,
+		} = this.props;
+
+		const mappedAttributes = { ...unMappedAttributes, url };
+		const setMappedAttributes = ( { url: href, ...restAttributes } ) =>
+			href === undefined
+				? setAttributes( restAttributes )
+				: setAttributes( { ...restAttributes, href } );
```

Comment:
The link settings sets the attribute named `url`, but Image block's `url` attribute is the URL of the image itself, and instead uses the attribute named `href` to represent the link URL.

Originally, to work around this, I used [dynamic properties](https://github.com/WordPress/gutenberg/pull/27292/commits/c7cf5873823a880866facc03e290e6fce9c3c7ad#diff-1c6b716518b93eef13a216ed21fa8d92122f8941fbac8378f13f73a4ae5314dc) in `LinkSettings` to allow the consumer to customize the attribute name to be set.

Here, I reverted those changes to `LinkSettings`, and instead map the attibutes in `ImageEdit`. As the link picker is used in more places, it might make sense to re-evaluate this pattern, depending on the commonality of the possible attribute names (e.g. `url`, `href`). WDYT?

Reply:
I like the dynamic properties approach. It seems fairly clean. The code ^ is a bit harder for me to understand. But I think this could be done as part of a larger refactor.

One thing that I noticed is that we always seem to set the options such as labels etc. 

I think we could really benefit from having a default name for labels. so that we only pass in the options that are different from the default.

Reply:
> I think we could really benefit from having a default name for labels. so that we only pass in the options that are different from the default.

Exactly my thoughts! I think we can refactor during the iteration for further consolidation [described here](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1655#issuecomment-745619398). I think with the refactor we should have the opportunity to both reduce the surface area of the interface, and also make the components appropriately general (in particular, decoupled from particular attribute names and even `setAttributes`).